pkg/binding/list_test.go

```diff
@@ -0,0 +1,244 @@
+package binding
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/devfile/library/pkg/devfile/parser"
+	"github.com/golang/mock/gomock"
+	"github.com/redhat-developer/odo/pkg/api"
+	"github.com/redhat-developer/odo/pkg/kclient"
+	bindingApis "github.com/redhat-developer/service-binding-operator/apis"
+	"github.com/redhat-developer/service-binding-operator/apis/binding/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var apiServiceBinding = api.ServiceBinding{
+	Name: "my-nodejs-app-cluster-sample",
+	Spec: api.ServiceBindingSpec{
+		Application: corev1.ObjectReference{
+			Name:       "my-nodejs-app-app",
+			APIVersion: deploymentApiVersion,
+			Kind:       deploymentKind,
+		},
+		Services: []corev1.ObjectReference{
+			{
+				Name:       "cluster-sample",
+				APIVersion: ClusterAPIVersion,
+				Kind:       clusterKind,
+			},
+		},
+		BindAsFiles:            true,
+		DetectBindingResources: true,
+	},
+}
+
+var bindingServiceBinding = v1alpha1.ServiceBinding{
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "my-nodejs-app-cluster-sample",
+		Labels: map[string]string{
+			"odo.dev/mode": "Dev",
+		},
+	},
+	Spec: v1alpha1.ServiceBindingSpec{
+		Application: v1alpha1.Application{
+			Ref: v1alpha1.Ref{
+				Group:   deploymentGVK.Group,
+				Version: deploymentGVK.Version,
+				Kind:    deploymentKind,
+				Name:    "my-nodejs-app-app",
+			},
+		},
+		Services: []v1alpha1.Service{
+			{
+				NamespacedRef: v1alpha1.NamespacedRef{
+					Ref: v1alpha1.Ref{
+						Group:   clusterGVK.Group,
+						Version: clusterGVK.Version,
+						Kind:    clusterKind,
+						Name:    "cluster-sample",
+					},
+				},
+			},
+		},
+		BindAsFiles:            true,
+		DetectBindingResources: true,
+	},
+	Status: v1alpha1.ServiceBindingStatus{
+		Conditions: []metav1.Condition{
+			{
+				Type:   bindingApis.InjectionReady,
+				Status: metav1.ConditionTrue,
+			},
+		},
+		Secret: "asecret",
+	},
+}
+
+var sbSecret = corev1.Secret{
+	Data: map[string][]byte{
+		"akey": []byte("avalue"),
+	},
+}
+
+func TestBindingClient_ListAllBindings(t *testing.T) {
+	type fields struct {
+		kubernetesClient func(ctrl *gomock.Controller) kclient.ClientInterface
+	}
+	type args struct {
+		devfileObj parser.DevfileObj
+		context    string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    []api.ServiceBinding
+		want1   []string
```

Comment:
Surely nitpicking, but could you rename this field name into something more meaningful? I had to look at the implementation in `BindingClient#ListAllBindings` to understand what was expected there..
Same thing in L231 with `got1`.